components/sup/src/http_gateway.rs

```diff
@@ -116,16 +117,22 @@ impl ToSocketAddrs for ListenAddr {
     }
 }
 
+impl fmt::Display for ListenAddr {
+    fn fmt(&self, f: &mut fmt::Formatter) -> result::Result<(), fmt::Error> {
+        write!(f, "{}", self.0)
+    }
+}
+
 struct ManagerState;
 
 impl typemap::Key for ManagerState {
     type Value = manager::State;
 }
 
-pub struct Server(Iron<Chain>);
+pub struct Server(Iron<Chain>, ListenAddr);
```

Comment:
This is owned by the manager struct (and should be) and should just be passed in by ref to `start()`. That'll also allow you to avoid the clone when starting and running the server at the start of the main loop in the manager